### PR TITLE
Improve capture handling

### PR DESCRIPTION
**Pull request by andersem**
- Use Dintero adapter for all API requests
- Perform capture, refund and void in one handler for both redirect and embedded
- Attempt capture even if amount does not match. This way we'll get notified.
- Change status to on-hold if capture failed
- Fix issue where we could not capture redirect-order if embed was active, and vice-versa.

### CODE REVIEW

**Review by chaikovskyia — APPROVED** [review]

File: includes/class-wc-gateway-dintero-hp.php [hunk 1]
```diff
@@ -1201,13 +1195,6 @@ private function check_capture( $order_id ) {
 			}
 			$transaction = self::_adapter()->get_transaction($transaction_id);
 
-
-			if ( ! array_key_exists( 'merchant_reference', $transaction ) ) {
-				$order->add_order_note(__('Could not capture transaction: Merchant reference id not found'));
-				$order->save_meta_data();
-				return false;
-			}
-
```
Comment (frodeaa) [inline includes/class-wc-gateway-dintero-hp.php:1210]: should we not add a note if capture is "aborted" because of missing merchant reference?

Comment (frodeaa) [reply includes/class-wc-gateway-dintero-hp.php:1210]: do we need to check the merchant_reference when the transaction is resolved from the order?
https://github.com/Dintero/Dintero.Checkout.WooCommerce/pull/96/files#diff-4aaca69ad3c4f3a1983beefb078889a21699f59e2f8473dbe000e11e4f594248L1215

File: includes/class-wc-gateway-dintero-hp.php [hunk 2]
```diff
@@ -1233,130 +1220,240 @@ private function capture( $order, $transaction = null ) {
...
-						'vat'         => $item_tax_percentage,
-						'amount'      => $item_line_total_amount,
-						'line_id'     => $productId
-					);
-					array_push( $items, $item );
-				}
+			if ($this->is_redirect_payment($transaction)) {
+				$items = $this->get_items_to_capture_redirect($order);
+			} else {
+				$items = $this->get_items_to_capture_embed($order);
```
Comment (frodeaa) [inline includes/class-wc-gateway-dintero-hp.php:1255]: how is items different when capturing redirect versus embedded?

**Review by frodeaa — APPROVED** [review]

File: includes/class-wc-gateway-dintero-hp.php (hunk 2 above)
Comment (andersem) [reply includes/class-wc-gateway-dintero-hp.php:1255]: For redirect, the id is like this: `item_1`, `item_2`, and the line_id is `1`, `2`, etc.

For embed, it's set to the product_id, which makes more sense.

In a new improvement, we can align the ids

**Comment by andersem** [comment]
Released in https://github.com/Dintero/Dintero.Checkout.WooCommerce/releases/tag/v2021.11.03

**Pull request merged**